### ISSUE TITLE
[java] ClassWithOnlyPrivateConstructorsShouldBeFinal should consider lombok's @NoArgsConstructor

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4199]
**Affects PMD Version:6.51.0**

**Rule: ClassWithOnlyPrivateConstructorsShouldBeFinal**

Please provide the rule name and a link to the rule documentation:
<https://pmd.sourceforge.io/pmd-6.51.0/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal>

**Description:**
In the following code example, the annotation `@NoArgsConstructor` can generate a public constructor `public`  and causes a false positive that PMD considers it is a class with only private constructors.

**Code Sample demonstrating the issue:**

```java
import lombok.NoArgsConstructor;
@NoArgsConstructor
public class Test {
    private Test(int a, int b, int c, String abc, long d, double p, String[] arr, int data, long in, float fl, String res) {}
}
```

**Expected outcome:**
PMD reports a violation at line 4, but that's wrong. That's a false positive.

**Running PMD through:** *CLI*

**Comment by adangel** [comment]
Duplicate of #4188
